drivers/oled/oled_driver.c

```diff
@@ -82,26 +86,58 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 // Charge Pump Commands
 #define CHARGE_PUMP 0x8D
 
+// Commands specific to the SH1107 chip
+#define SH1107_DISPLAY_START_LINE 0xDC
+#define SH1107_MEMORY_MODE_PAGE 0x20
+#define SH1107_MEMORY_MODE_VERTICAL 0x21
+
 // Misc defines
 #ifndef OLED_BLOCK_COUNT
 #    define OLED_BLOCK_COUNT (sizeof(OLED_BLOCK_TYPE) * 8)
 #endif
 #ifndef OLED_BLOCK_SIZE
 #    define OLED_BLOCK_SIZE (OLED_MATRIX_SIZE / OLED_BLOCK_COUNT)
 #endif
+// Default display clock
+#if !defined(OLED_DISPLAY_CLOCK)
+#    define OLED_DISPLAY_CLOCK 0x80
+#endif
+// Default VCOMH deselect value
+#if !defined(OLED_VCOM_DETECT)
+#    define OLED_VCOM_DETECT 0x20
+#endif
+#if !defined(OLED_PRE_CHARGE_PERIOD)
+#    define OLED_PRE_CHARGE_PERIOD 0xF1
+#endif
+
 
 #define OLED_ALL_BLOCKS_MASK (((((OLED_BLOCK_TYPE)1 << (OLED_BLOCK_COUNT - 1)) - 1) << 1) | 1)
 
+#define OLED_IC_HAS_HORIZONTAL_MODE (OLED_IC == OLED_IC_SSD1306)
+#define OLED_IC_COM_PINS_ARE_COLUMNS (OLED_IC == OLED_IC_SH1107)
+
+#ifndef OLED_COM_PIN_COUNT
+#    if OLED_IC == OLED_IC_SSD1306
+#        define OLED_COM_PIN_COUNT 64
+#    elif OLED_IC == OLED_IC_SH1106
+#        define OLED_COM_PIN_COUNT 64
+#    elif OLED_IC == OLED_IC_SH1107
+#        define OLED_COM_PIN_COUNT 128
+#    else
+#        error Invalid OLED_IC value
+#    endif
+#endif
+
+#ifndef OLED_COM_PIN_OFFSET
+#    define OLED_COM_PIN_OFFSET 0
+#endif
+
 // i2c defines
 #define I2C_CMD 0x00
 #define I2C_DATA 0x40
-#if defined(__AVR__)
-#    define I2C_TRANSMIT_P(data) i2c_transmit_P((OLED_DISPLAY_ADDRESS << 1), &data[0], sizeof(data), OLED_I2C_TIMEOUT)
-#else // defined(__AVR__)
-#    define I2C_TRANSMIT_P(data) i2c_transmit((OLED_DISPLAY_ADDRESS << 1), &data[0], sizeof(data), OLED_I2C_TIMEOUT)
-#endif // defined(__AVR__)
-#define I2C_TRANSMIT(data) i2c_transmit((OLED_DISPLAY_ADDRESS << 1), &data[0], sizeof(data), OLED_I2C_TIMEOUT)
-#define I2C_WRITE_REG(mode, data, size) i2c_writeReg((OLED_DISPLAY_ADDRESS << 1), mode, data, size, OLED_I2C_TIMEOUT)
+
+#undef ARRAY_SIZE
+#define ARRAY_SIZE(arr) sizeof(arr) / sizeof(arr[0])
```

Comment:
Does the version of `ARRAY_SIZE()` from `quantum/util.h` fail here for some reason?

Reply:
left overs removed